src/viam/sessions_client.py

```diff
@@ -0,0 +1,145 @@
+import asyncio
+from datetime import timedelta
+from typing import Optional
+
+from grpclib import Status
+from grpclib.client import Channel
+from grpclib.events import RecvTrailingMetadata, SendRequest, listen
+from grpclib.exceptions import GRPCError, StreamTerminatedError
+from grpclib.metadata import _MetadataLike
+
+from viam import _TASK_PREFIX, logging
+from viam.proto.robot import RobotServiceStub, SendSessionHeartbeatRequest, StartSessionRequest, StartSessionResponse
+
+LOGGER = logging.getLogger(__name__)
+SESSION_METADATA_KEY = "viam-sid"
+
+EXEMPT_METADATA_METHODS = frozenset(
+    [
+        "/grpc.reflection.v1alpha.ServerReflection/ServerReflectionInfo",
+        "/proto.rpc.webrtc.v1.SignalingService/Call",
+        "/proto.rpc.webrtc.v1.SignalingService/CallUpdate",
+        "/proto.rpc.webrtc.v1.SignalingService/OptionalWebRTCConfig",
+        "/proto.rpc.v1.AuthService/Authenticate",
+        "/viam.robot.v1.RobotService/ResourceNames",
+        "/viam.robot.v1.RobotService/ResourceRPCSubtypes",
+        "/viam.robot.v1.RobotService/StartSession",
+        "/viam.robot.v1.RobotService/SendSessionHeartbeat",
+    ]
+)
+
+
+async def delay(coro, seconds):
```

Comment:
there's a `loop.call_later` function that might be useful here:
```python3
loop = asyncio.get_running_loop()
loop.call_later(seconds, coro)
```

Reply:
This looked promising, but unfortunately `loop.call_later` takes a sync callback and not a co-routine: https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.call_later